### PR TITLE
Exclusion distance correction

### PR DESCRIPTION
**Pull request by ETVincent**
There was a bug for the PyGRB analysis wherein the 50% exclusion distance was not reported accurately in the table. You can see an example of the issue here: [https://ldas-jobs.ligo.caltech.edu/~francesco.pannarale/LVC/pygrb_sep2024_2/6._open_box/6.02_exclusion_distances/](https://ldas-jobs.ligo.caltech.edu/~francesco.pannarale/LVC/pygrb_sep2024_2/6._open_box/6.02_exclusion_distances/) wherein the reported 50% exclusion distance does not match that from the plot. 

## Standard information about the request

This is a bug fix to correct the issue. 

This change affects: PyGRB.

## Contents
It appears as though there was some extra duplicated code. This extra code was removed. 

Additionally, a variable was converted from an int into a dictionary to hold the information better. 

## Testing performed
During testing, the reported 50% exclusion distance was closer to that from the plot. 

- [x] The author of this pull request confirms they will adhere to the [code of conduct](https://github.com/gwastro/pycbc/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: bin/pygrb/pycbc_pygrb_efficiency [hunk 1]
```diff
@@ -597,6 +578,8 @@ yerr_low, yerr_high, fraction_mc = \
 red_efficiency = (fraction_mc) - (yerr_low) * scipy.stats.norm.isf(0.1)
 
 # Calculate and save to disk 50% and 90% exclusion distances
+# excl_dist dictionary contains 50% and 90% exclusion distances 
+excl_dist = {}
```
Comment (pannarale) [inline bin/pygrb/pycbc_pygrb_efficiency:582]: It looks like you can use this dictionary only, and remove `excl_dist_dict`.  Details below.

File: bin/pygrb/pycbc_pygrb_efficiency [hunk 2]
```diff
@@ -611,13 +594,13 @@ for percentile in [50, 90]:
         d_low = dist_plot_vals[i-1]
         e = excl_efficiency[i]
         e_low = excl_efficiency[i-1]
-        excl_dist = d + (e - (percentile / 100.)) * (d - d_low) /\
+        excl_dist[percentile] = d + (e - (percentile / 100.)) * (d - d_low) /\
```
Comment (pannarale) [inline bin/pygrb/pycbc_pygrb_efficiency:597]:
```suggestion
        excl_dist[f"{percentile}%"] = d + (e - (percentile / 100.)) * (d - d_low) /\
```

File: bin/pygrb/pycbc_pygrb_efficiency [hunk 3]
```diff
@@ -611,13 +594,13 @@ for percentile in [50, 90]:
...
             (e_low - e)
     else:
         # Warn the user if the exclusion distance cannot be established,
         # but let the workflow continue: the user will see the plot(s) and
         # repeat with more injections
-        excl_dist = 0
+        excl_dist[percentile] = 0
```
Comment (pannarale) [inline bin/pygrb/pycbc_pygrb_efficiency:603]:
```suggestion
        excl_dist[f"{percentile}%"] = 0
```

File: bin/pygrb/pycbc_pygrb_efficiency [hunk 4]
```diff
@@ -627,8 +610,8 @@ if opts.exclusion_dist_output_file:
     excl_dist_dict = {}
     excl_dist_dict['inj_set'] = inj_set_name
     excl_dist_dict['trial_name'] = opts.trial_name
-    excl_dist_dict['50%'] = sens_dist
-    excl_dist_dict['90%'] = excl_dist
+    excl_dist_dict['50%'] = excl_dist[50]
+    excl_dist_dict['90%'] = excl_dist[90]
```
Comment (pannarale) [inline bin/pygrb/pycbc_pygrb_efficiency:614]:
```suggestion
    excl_dist['inj_set'] = inj_set_name
    excl_dist['trial_name'] = opts.trial_name
```

File: bin/pygrb/pycbc_pygrb_efficiency [hunk 5]
```diff
@@ -659,15 +642,15 @@ if opts.onsource_output_file:
...
     ax.set_ylim([0, 1])
     ax.set_xlim(0, 2.*upper_dist - lower_dist)
     plot_title = "Exclusion distance - "+inj_set_name
     plot_caption = "Injection recovery efficiency using "
     plot_caption += "BestNR as detection statistic.  "
     plot_caption += "Injections louder than loudest foreground trigger.\n"
-    plot_caption += f" 90%% exclusion distance: {excl_dist} Mpc\n"
-    plot_caption += f" 50%% sensitive distance: {sens_dist} Mpc"
+    plot_caption += f" 90%% exclusion distance: {excl_dist[90]} Mpc\n"
+    plot_caption += f" 50%% sensitive distance: {excl_dist[50]} Mpc"
```
Comment (pannarale) [inline bin/pygrb/pycbc_pygrb_efficiency:653]:
```suggestion
    plot_caption += f" 90%% exclusion distance: {excl_dist['90%']} Mpc\n"
    plot_caption += f" 50%% sensitive distance: {excl_dist['50%']} Mpc"
```

**Review by pannarale — CHANGES_REQUESTED** [review]

File: bin/pygrb/pycbc_pygrb_efficiency (hunk 4 above)
Comment (ETVincent) [reply bin/pygrb/pycbc_pygrb_efficiency]: Completed in update below

**Review by pannarale — APPROVED** [review]

**Pull request merged**